app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -709,14 +709,35 @@ class NimbusExperimentCloneSerializer(
         slug_field="slug", queryset=NimbusExperiment.objects.all()
     )
     name = serializers.CharField(min_length=1, max_length=80, required=True)
+    rollout_branch_slug = serializers.CharField(required=False, allow_null=True)
 
     class Meta:
         model = NimbusExperiment
         fields = (
             "parent_slug",
             "name",
+            "rollout_branch_slug",
         )
 
+    def validate(self, data):
+        data = super().validate(data)
+        rollout_branch_slug = data.get("rollout_branch_slug", None)
+        if rollout_branch_slug:
+            parent = data.get("parent_slug")
+            if not parent.branches.filter(slug=rollout_branch_slug).exists():
+                raise serializers.ValidationError(
+                    {
+                        "rollout_branch_slug": (
+                            f"Rollout branch {rollout_branch_slug} does not exist."
+                        )
+                    }
+                )
+        return data
```

Comment:
Oh yeah yep.  I was trying to figure out if we could just use the builtin validation of the slug related field, but we'd have to subclass it and pass in the parent in init and override `get_queryset` so... i'm not sure it's any simpler than this.  But yep this looks good 👍